test/mpi/coll/test_coll_algos.sh

```diff
@@ -30,6 +30,12 @@ for algo_name in ${algo_names}; do
     done
 done
 
+#test scatter_recexch_allgather algorithm
+algo_name="scatter_recexch_allgather"
```

Comment:
Why not integrate this into the above loop? Is it just because we would end up setting a lot of environment variables?

Reply:
Yes, we will end up setting a lot of unused variables for other algorithm tests. If you are fine with that, I can put it back in the loop.

Reply:
Yes, let's do that. The one off stuff, at least IMO, makes it harder to understand.

Reply:
Done. I don't think we need to rerun the tests.